ndbench-dynamodb-plugins/src/main/java/com/netflix/ndbench/plugin/dynamodb/DynamoDBKeyValue.java

```diff
@@ -0,0 +1,233 @@
+import com.amazonaws.services.dynamodbv2.model.GlobalTable;
+import com.amazonaws.services.dynamodbv2.model.KeySchemaElement;
+import com.amazonaws.services.dynamodbv2.model.KeyType;
+import com.amazonaws.services.dynamodbv2.model.ProvisionedThroughput;
+import com.amazonaws.services.dynamodbv2.model.ScalarAttributeType;
+import com.amazonaws.services.dynamodbv2.model.TableDescription;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import com.netflix.ndbench.api.plugin.DataGenerator;
+import com.netflix.ndbench.api.plugin.NdBenchClient;
+import com.netflix.ndbench.api.plugin.annotations.NdBenchClientPlugin;
+import com.netflix.ndbench.api.plugin.common.NdBenchConstants;
+import com.netflix.ndbench.plugin.dynamodb.configs.DynamoDBConfigs;
+
+/**
+ * This NDBench plugin provides a single key value for AWS DynamoDB.
+ * 
+ * @author ipapapa
+ */
+@Singleton
+@NdBenchClientPlugin("DynamoDBKeyValue")
+public class DynamoDBKeyValue implements NdBenchClient {
+	private final Logger logger = LoggerFactory.getLogger(DynamoDBKeyValue.class);
+	private static AmazonDynamoDB client;
+	private static DynamoDB dynamoDB;
+	private static AWSCredentialsProvider awsCredentialsProvider;
+	private DynamoDBConfigs config;
+	private static Table table;
+
+	private DataGenerator dataGenerator;
+
+	/**
+	 * Credentials will be loaded based on the environment. In AWS, the credentials
+	 * are based on the instance. In a local deployment they will have to provided.
+	 */
+	@Inject
+	public DynamoDBKeyValue(AWSCredentialsProvider credential, DynamoDBConfigs config, DataGenerator dataGenerator) {
+		this.config = config;
+		if (System.getenv(NdBenchConstants.DISCOVERY_ENV).equals("AWS")) {
+			awsCredentialsProvider = credential;
+		} else {
+			/*
+			 * The ProfileCredentialsProvider will return your [default] credential profile
+			 * by reading from the credentials file located at
+			 * (/home/username/.aws/credentials).
+			 */
+			awsCredentialsProvider = new ProfileCredentialsProvider();
+			try {
+				awsCredentialsProvider.getCredentials();
+			} catch (Exception e) {
+				throw new AmazonClientException("Cannot load the credentials from the credential profiles file. "
+						+ "Please make sure that your credentials file is at the correct "
+						+ "location (/home/<username>/.aws/credentials), and is in valid format.", e);
+			}
+		}
+	}
+
+	@Override
+	public void init(DataGenerator dataGenerator) throws Exception {
+		logger.info("Initing DynamoDB plugin");
+		client = AmazonDynamoDBClientBuilder.standard().withCredentials(awsCredentialsProvider).build();
+		dynamoDB = new DynamoDB(client);
+
+		/*
+		 * Create a table with a primary hash key named 'name', which holds a string.
+		 * Several properties such as provisioned throughput and atribute names are
+		 * defined in the configuration interface.
+		 */
+
+		logger.debug("Creating table if it does not exist yet");
+		table = dynamoDB.createTable(this.config.getTableName(),
```

Comment:
what happens when the table already exists? does it fail or is ignored?

Reply:
```
ResourceInUseException
The operation conflicts with the resource's availability. For example, you attempted to recreate an existing table, or tried to delete a table currently in the CREATING state.

HTTP Status Code: 400
```

Resource: https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_CreateTable.html